client/login/magic-login/gravatar/index.tsx

```diff
@@ -0,0 +1,1038 @@
+						{ translate( 'Continue' ) }
+					</FormButton>
+				</form>
+				<footer
+					className={ clsx( 'grav-powered-magic-login__footer', {
+						'grav-powered-magic-login__footer--vertical': ! isFromGravatar3rdPartyApp,
+					} ) }
+				>
+					<button
+						onClick={ () => {
+							handleGravPoweredEmailCodeSend( usernameOrEmail );
+
+							recordTracksEvent( 'calypso_gravatar_powered_magic_login_click_resend_email', {
+								type: 'code',
+								client_id: oauth2ClientId,
+								client_name: oauth2Client?.title,
+							} );
+						} }
+						disabled={ Boolean( isRequestingEmail ) || Boolean( resendEmailCountdown ) }
+					>
+						{ resendEmailCountdown === 0
+							? translate( 'Send again' )
+							: translate( 'Send again (%(countdown)d)', {
+									args: { countdown: resendEmailCountdown },
+							  } ) }
+					</button>
+					{ shouldShowSwitchEmail && (
+						<button
+							onClick={ () => {
+								resetResendEmailCountdown();
+								handleGravPoweredEmailSwitch();
+							} }
+						>
+							{ translate( 'Switch email' ) }
+						</button>
+					) }
+					<a href="https://gravatar.com/support" target="_blank" rel="noreferrer">
+						{ translate( 'Need help logging in?' ) }
+					</a>
+				</footer>
+			</div>
+		);
+	};
+
+	const GravPoweredEmailLinkVerification = () => {
+		const emailAddress =
+			usernameOrEmail && usernameOrEmail.includes( '@' ) ? usernameOrEmail : null;
+		const isSendingEmail = useSelector( isFetchingMagicLoginEmail );
+
+		const emailTextOptions = {
+			components: {
+				sendEmailButton: (
+					<button
+						onClick={ () => {
+							usernameOrEmail &&
+								dispatch(
+									sendEmailLogin( usernameOrEmail, {
+										redirectTo: currentQueryArguments?.redirect_to ?? '', // This is a required field
+										requestLoginEmailFormFlow: true,
+										createAccount: true,
+										flow: oauth2Client ? getGravatarOAuth2Flow( oauth2Client ) ?? '' : '',
+										showGlobalNotices: true,
+										/**
+										 * The below are required fields but not used in this flow
+										 */
+										tokenType: 'link',
+										source: false,
+										blogId: '',
+										loginFormFlow: false,
+										isMobileAppLogin: false,
+									} )
+								);
+
+							recordTracksEvent( 'calypso_gravatar_powered_magic_login_click_resend_email', {
+								type: 'link',
+								client_id: oauth2ClientId,
+								client_name: oauth2Client?.title,
+							} );
+						} }
+						disabled={ isSendingEmail }
+					/>
+				),
+				showMagicLoginButton: (
+					<button
+						className="grav-powered-magic-login__show-magic-login"
+						onClick={ () => {
+							resetResendEmailCountdown();
+							handleGravPoweredEmailSwitch();
+						} }
+					/>
+				),
+			},
+		};
+
+		return (
+			<div className="grav-powered-magic-login__content">
+				<GravatarLoginLogo
+					iconUrl={ oauth2Client?.icon }
+					alt={ oauth2Client?.title ?? '' }
+					isCoBrand={ isGravatarFlow }
+				/>
```

Comment:
This being repeated isn't great. We should extract this out and wrap these components with (along with the notice and locale-suggestions). But this was the original way that's being ported into here; it's just a little more pronounced now.